File: Client/src/app/modules/core/models/shipmentUpdateDTO.ts
import { ProductShipmentDto } from './productShipmentDto';

export interface ShipmentUpdateDTO { 
    id: number;
    clientName?: string;
    price?: number;
    note?: string;
    addressId?: number;
    locationId?: number;
    street?: string;
    number?: string;
    state?: number;
    date?: Date;
    products?: Array<ProductShipmentDto>;
}

File: Api/FormulasApi/DTOS/ShipmentUpdateDTO.cs
using System.ComponentModel.DataAnnotations;

namespace FormulasApi.DTOS
{
    public class ShipmentUpdateDTO
    {
        [Required(ErrorMessage = "El campo id es obligatorio.")]
        public int Id { get; set; }
        public string ClientName { get; set; }
        public float Price { get; set; }
        public string Note { get; set; } 

        public int AddressId { get; set; }  
        public int LocationId { get; set; }
        public string Street { get; set; }
        public string Number { get; set; }

        public int? State { get; set; }
        public DateTime Date { get; set; }
        public List<ProductShipmentDto> Products { get; set; } = new List<ProductShipmentDto>();
    }
}
